app/models/site_setting.rb

```diff
@@ -16,6 +16,8 @@ class SiteSetting < ActiveRecord::Base
       elsif self.data_type == SiteSettings::TypeSupervisor.types[:uploaded_image_list]
         upload_ids = self.value.split("|").compact.uniq
         UploadReference.ensure_exist!(upload_ids: upload_ids, target: self)
+      elsif self.data_type == SiteSettings::TypeSupervisor.types[:emoji_list]
+        Emoji.clear_cache && Discourse.request_refresh!
```

Comment:
One small request here, I think we can just move this to https://github.com/discourse/discourse/blob/main/config/initializers/014-track-setting-changes.rb